core/src/apps/bitcoin/sign_tx/layout.py

```diff
@@ -54,50 +54,39 @@ async def confirm_output(
         text = Text("Confirm sending", ui.ICON_SEND, ui.GREEN)
         text.normal(format_coin_amount(output.amount, coin) + " to")
         text.mono(*split_address(address_short))
-    return await confirm(ctx, text, ButtonRequestType.ConfirmOutput)
+    await require_confirm(ctx, text, ButtonRequestType.ConfirmOutput)
 
 
 async def confirm_total(
     ctx: wire.Context, spending: int, fee: int, coin: coininfo.CoinInfo
-) -> bool:
+) -> None:
     from trezor.ui.text import Text
-    from apps.common.confirm import hold_to_confirm
+    from apps.common.confirm import require_hold_to_confirm
 
     text = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
     text.normal("Total amount:")
     text.bold(format_coin_amount(spending, coin))
     text.normal("including fee:")
     text.bold(format_coin_amount(fee, coin))
-    return await hold_to_confirm(ctx, text, ButtonRequestType.SignTx)
+    await require_hold_to_confirm(ctx, text, ButtonRequestType.SignTx)
 
 
 async def confirm_feeoverthreshold(
     ctx: wire.Context, fee: int, coin: coininfo.CoinInfo
-) -> bool:
+) -> None:
     from trezor.ui.text import Text
-    from apps.common.confirm import confirm
+    from apps.common.confirm import require_confirm
 
     text = Text("High fee", ui.ICON_SEND, ui.GREEN)
     text.normal("The fee of")
     text.bold(format_coin_amount(fee, coin))
     text.normal("is unexpectedly high.", "Continue?")
-    return await confirm(ctx, text, ButtonRequestType.FeeOverThreshold)
+    await require_confirm(ctx, text, ButtonRequestType.FeeOverThreshold)
 
 
-async def confirm_foreign_address(
```

Comment:
was this function unused?

Reply:
Yes AFAICT. We are using `paths.show_path_warning` in `__init__` instead.